src/renderers/WebGLRenderer.js

```diff
@@ -701,13 +708,14 @@ function WebGLRenderer( parameters ) {
 		state.setMaterial( material, frontFaceCW );
 
 		var program = setProgram( camera, fog, material, object );
-		var geometryProgram = geometry.id + '_' + program.id + '_' + ( material.wireframe === true );
 
 		var updateBuffers = false;
 
-		if ( geometryProgram !== _currentGeometryProgram ) {
+		if ( geometry.id !== _currentGeometryProgram.geometryID || program.id !== _currentGeometryProgram.programID || material.wireframe !== _currentGeometryProgram.wireframe ) {
```

Comment:
Similarly `wireframe` comparison should be?

     _currentGeometryProgram.wireframe !== ( material.wireframe === true )

Reply:
Are you saying that checking `true !== undefined` is slower than `true !== false`?

Reply:
Nope. _currentGeometryProgram.wireframe is true/false but material.wireframe can be undefined because some Materials don't have .wireframe, then undefined !== false will be false.

Reply:
Oh...

Reply:
Fixed.